### PR TITLE
provide generic community docs

### PR DESCRIPTION
**Pull request by pearsonca**
Some very mild changes to be generic across epinowcast ...-verse.

### CODE REVIEW

File: STYLE_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# Style Guide
+
+This document outlines the style guide for the `epinowcast` package. This guide is a work in progress and will be updated as the package evolves. We welcome contributions to this guide and encourage you to raise issues or submit PRs if you have any suggestions.
+
+In addition to this guide we also follow the [tidyverse style guide](https://style.tidyverse.org/). This guide is a subset of the `tidyverse` style guide and outlines the additional style requirements for the `epinowcast` package.
+
+## Naming conventions
+
+- We use a `enw_` prefix to delineate functions that are exported by the `epinowcast` package. This is to avoid conflicts with other packages and to make it clear to users which functions are part of the package.
```
Comment (seabbs) [inline STYLE_GUIDE.md]: This isn't portable to other packages.

**Review by seabbs — APPROVED** [review]
This is nice - thanks. It feels like maybe the style guide is not portable as is so we should either leave out or substantially reduce it

File: STYLE_GUIDE.md (hunk 1 above)
Comment (pearsonca) [reply STYLE_GUIDE.md]: good catch, was too breezy in looking at this one.

that said, i think with some adjustments, could be made into an epinowcast-a-rama (...refusing to give into -verse nomenclature) general doc, which then specific projects have their own additions to (e.g. the `enw_` prefix).

Comment (seabbs) [reply STYLE_GUIDE.md]: It definitely could but maybe we should make an issue + PR pair for that and deal with on its own?

File: STYLE_GUIDE.md [hunk 2]
```diff
@@ -25,34 +30,18 @@ More generally when adding functions from external packages (i.e. even if they a
...
+- Any required inputs should be clearly documented in the function documentation, particularly in terms of type, but also other constraints (e.g. presence/absence of columns).
+- Any expressed constraint in exported functions should be verified using some sort of `check_` expression, and ideally unit tests written correspondingly to confirm that `check_` rejects bad input.
+- Many of methods across the `epinowcast` packages work with `data.frame` inputs (and subclasses of `data.frame` like `data.table` and `tibble`). Internally, for performance and syntax reasons, we prefer to use `data.table`s explicitly and that is generally the type returned by functions. However, we will continue to accept `data.frame`-like arguments as inputs.
+- Translation from `data.frame` to `data.table` can be handled by `coerce_dt`, `check_dt` or `make_dt` (`coerce_dt` and `check_dt` combined) which will be provided in a to-be-released package `make_dt`
```
Comment (seabbs) [inline STYLE_GUIDE.md:36]: In other packages not written by us I think we should be open to a dplyr workflow. This sounds a little like we are not but I think we can probably cross that bridge when it comes.

**Review by seabbs — APPROVED** [review]
Nice update. This is good to go. See comment as food for thought on other community contributors using `dplyr` etc workflows in their packages (not a bridge that we have had to cross yet).

**Pull request merged**